src/map/utils/itemutils.cpp

```diff
@@ -472,15 +482,27 @@ namespace itemutils
                     g_pDropList[DropID] = new DropList_t;
                 }
 
-                DropItem_t DropItem;
+                DropList_t* dropList = g_pDropList[DropID];
 
-                DropItem.ItemID  = (uint16)Sql_GetIntData(SqlHandle,1);
-                DropItem.DropType = (uint8)Sql_GetIntData(SqlHandle,2);
-                DropItem.DropRate = (uint16)Sql_GetIntData(SqlHandle,3);
-                DropItem.GroupId = (uint8)Sql_GetIntData(SqlHandle,4);
-                DropItem.GroupRate = (uint16)Sql_GetIntData(SqlHandle,5);
+                uint16 ItemID = (uint16)Sql_GetIntData(SqlHandle, 1);
+                uint8 DropType = (uint8)Sql_GetIntData(SqlHandle, 2);
+                uint16 DropRate = (uint16)Sql_GetIntData(SqlHandle, 3);
 
-                g_pDropList[DropID]->push_back(DropItem);
+                if (DropType == DROP_GROUPED)
+                {
+                    uint8 GroupId = (uint8)Sql_GetIntData(SqlHandle, 4);
+                    uint16 GroupRate = (uint16)Sql_GetIntData(SqlHandle, 5);
+                    //This requires the database to have the groups in order
```

Comment:
should this silently fail if it isn't?

Reply:
Yeah, I was originally going to do that then I was wondering if there is any sort of validation on the database elsewhere. I think the best solution would be to have some test scripts that validate the database.

Assuming that you always use all of the groupids from 0 then it could emplace_back groups until it gets to the correct groupid with a while loop. I meant to make this change but couldn't remember it before...